src/main/java/com/google/firebase/auth/ProviderConfig.java

```diff
@@ -60,6 +62,24 @@ static void assertValidUrl(String url) throws IllegalArgumentException {
     }
   }
 
+  static List<Object> getNestedList(Map<String, Object> outerMap, String id) {
+    List<Object> list = (List<Object>) outerMap.get(id);
+    if (list == null) {
+      list = new ArrayList<Object>();
+      outerMap.put(id, list);
+    }
+    return list;
+  }
+
+  static Map<String, Object> getNestedMap(Map<String, Object> outerMap, String id) {
```

Comment:
I think `ensureNestedMap()` and `ensureNestedList()` indicate the intent of these a bit better.

Reply:
Done.